### DISCUSSION TITLE
Cannot download prisma engines while prisma generate due to 404 Error

### DISCUSSION THREAD

**Discussion by kritsanapr** [prisma/prisma#24019] [Q&A] [answered]
### Bug description

We got same error from both prisma generate and prisma version

```
root@0abec131dcb6:/workspaces/DPT-CM-MONOREPO-NEW-V2# prisma version 
prisma:warn Prisma only officially supports Linux on amd64 (x86_64) and arm64 (aarch64) system architectures. If you are using your own custom Prisma engines, you can ignore this warning, as long as you've compiled the engines for your system architecture "armv7l".
> Downloading Prisma engines for Node-API for linux-arm-openssl-3.0.x [                    ] 0%Error: Failed to fetch sha256 checksum at https://binaries.prisma.sh/all_commits/b9a39a7ee606c28e3455d0fd60e78c3ba82b1a2b/linux-arm-openssl-3.0.x/schema-engine.gz.sha256 - 404 Not Found

If you need to ignore this error (e.g. in an offline environment), set the PRISMA_ENGINES_CHECKSUM_IGNORE_MISSING environment variable to a truthy value.
Example: PRISMA_ENGINES_CHECKSUM_IGNORE_MISSING=1
```

OS: Docker node:22.0.0 on Mac M1 Chip with Sonoma 14.4.1 
Node.js version: v22.0.0

### Expected behavior

Should be able to show a version of prisma.

### Prisma information

```prisma
// Add your schema.prisma
generator client {
  provider        = "prisma-client-js"
  binaryTargets   = ["native", "darwin-arm64", "linux-musl", "debian-openssl-3.0.x", "linux-arm64-openssl-3.0.x"]
  previewFeatures = ["orderByNulls"]
}
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: MacOS Somona 14.4.1 (Apple Silicon)
- Node.js version: Docker node:22.0.0

### Prisma Version

```
 "prisma": "5.13.0",
```

**Comment by RubenNL** [comment]
Same on ARM64 (AWS t4g machine) with Node 22. Seems like the common variable is the node version, and some variant of ARM.

```
prisma:warn Prisma only officially supports Linux on amd64 (x86_64) and arm64 (aarch64) system architectures. If you are using your own custom Prisma engines, you can ignore this warning, as long as you've compiled the engines for your system architecture "aarch64".
Error: Failed to fetch sha256 checksum at https://binaries.prisma.sh/all_commits/b9a39a7ee606c28e3455d0fd60e78c3ba82b1a2b/linux-arm-openssl-3.0.x/libquery_engine.so.node.gz.sha256 - 404 Not Found

**Reply by janpio** [reply]
Are you also getting this warning first @RubenNL?
```
prisma:warn Prisma only officially supports Linux on amd64 (x86_64) and arm64 (aarch64) system architectures. If you are using your own custom Prisma engines, you can ignore this warning, as long as you've compiled the engines for your system architecture "armv7l".
```

**Reply by RubenNL** [reply]
This is the exact log:
```
prisma:warn Prisma only officially supports Linux on amd64 (x86_64) and arm64 (aarch64) system architectures. If you are using your own custom Prisma engines, you can ignore this warning, as long as you've compiled the engines for your system architecture "aarch64".
Error: Failed to fetch sha256 checksum at https://binaries.prisma.sh/all_commits/b9a39a7ee606c28e3455d0fd60e78c3ba82b1a2b/linux-arm-openssl-3.0.x/libquery_engine.so.node.gz.sha256 - 404 Not Found
If you need to ignore this error (e.g. in an offline environment), set the PRISMA_ENGINES_CHECKSUM_IGNORE_MISSING environment variable to a truthy value.
Example: PRISMA_ENGINES_CHECKSUM_IGNORE_MISSING=1
```

I just saw that I'm running on `node:22-alpine` image. With `node:21-alpine` i'm not getting this error.

I will post a debug log in the comment below.

**Comment by janpio** [comment]
It seems Prisma for some reason is trying to download a `linux-arm` engine file, which does not exist. As the warning message tells you, we only build for `linux-arm64`, which is why the download of this file returns a 404 and fails.

We'll now need to figure out why it is trying to download that, and if `arm64` would actually work on your systems.

**Reply by janpio** [reply]
@RubenNL @kritsanapr Can you set the env var `DEBUG=*` and run the commands again and share the full output you are getting? That might help us here. Thanks.

**Reply by RubenNL** [reply]
I'm not getting the error anymore, so this might be solved? I'm testing some more and will report back if it fails again.

**Reply by janpio** [reply]
See here @RubenNL: https://github.com/prisma/prisma/discussions/24019#discussioncomment-9274230

**Reply by kritsanapr** [reply]
@janpio I didn't find any errors. Trying the recommended method.

**Comment by aqrln** [comment, marked as answer]
This was an upstream issue which resulted in a Docker image for incompatible platform being downloaded (https://github.com/nodejs/docker-node/issues/2074). This is now fixed upstream, just make sure to pull new images:

```
docker pull node:22 --platform linux/arm64
```
(replace `node:22` with the tag you use)

Thanks for reporting this!

**Reply by kritsanapr** [reply]
It works!. Thank you very much.
